flink-table/flink-table-planner/src/main/java/org/apache/flink/table/planner/delegation/DialectFactory.java

```diff
@@ -62,5 +77,20 @@ public CatalogManager getCatalogManager() {
         public PlannerContext getPlannerContext() {
             return plannerContext;
         }
+
+        @Override
+        public Executor getExecutor() {
+            return executor;
+        }
+    }
+
+    /** Default implementation for {@link OperationExternalExecutor}. */
+    class DefaultOperationExternalExecutor implements OperationExternalExecutor {
```

Comment:
The default implementation can be `EmptyOperationExecutor` with comment: a default implementation of `ExtendedOperationExecutor` that doesn't extend any operation behavior but forward all operations to the Flink planner.